logstash-core/src/main/java/org/logstash/plugins/factory/Metrics.java

```diff
@@ -0,0 +1,61 @@
+package org.logstash.plugins.factory;
+
+import org.jruby.*;
+import org.jruby.anno.JRubyClass;
+import org.jruby.anno.JRubyMethod;
+import org.jruby.runtime.ThreadContext;
+import org.jruby.runtime.builtin.IRubyObject;
+import org.logstash.RubyUtil;
+import org.logstash.instrument.metrics.AbstractMetricExt;
+import org.logstash.instrument.metrics.AbstractNamespacedMetricExt;
+import org.logstash.instrument.metrics.MetricKeys;
+import org.logstash.instrument.metrics.NullMetricExt;
+
+import java.util.Arrays;
+
+@JRubyClass(name = "PluginMetricFactory")
+public final class Metrics extends RubyBasicObject {
```

Comment:
why are the Ruby/Java names so different, does it make sense?

Reply:
Should be named PluginMetricFactoryExt

Reply:
Done